lib/module.js

```diff
@@ -71,10 +66,10 @@ function readPackage(requestPath) {
     return packageMainCache[requestPath];
   }
 
-  try {
-    var jsonPath = path.resolve(requestPath, 'package.json');
-    var json = fs.readFileSync(jsonPath, 'utf8');
-  } catch (e) {
+  var jsonPath = path.resolve(requestPath, 'package.json');
+  var json = internalModuleReadFile(jsonPath);
```

Comment:
I don't think that should be a concern, especially not when there's potential perf gains. Nobody should depend on the exact stack trace.